### DISCUSSION TITLE
Computing spatial averages on the fly

### DISCUSSION THREAD

**Discussion by makrandak** [Nek5000/nekRS#555] [Q&A]
Hello, 

I am trying to compute x and z averages (imagine a statistically stationary channel flow periodic in x and z) of velocities at every timestep.  I further want to do the following calculation at the end of each timestep, 

u_new = u (solution at the end of the timestep) + ( u_avg_xz (calculated above and a function of y) - u_avg(y) ); 

"u_new" becomes the new solution, "u_avg(y)" is some analytical function I can evaluate.  I am thinking of using the limiter kernel in k-tau implementation for this. But I am stuck at getting the "u_avg_xz (calculated above and a function of y)" on the fly on GLL points/nodes to do the above calculation. 

I was looking at the `tavg.cpp` plugin (I am assuming they are running time averages), and I am wondering if I can use those kernels (modifying them) to do the averaging calculation I want. 
If there is any documentation available for doing such averages I would be very grateful if you point me to it. 

With sincere appreciation for your help,
Makrand

**Comment by stgeke** [comment]
Look for `planarAvg` e.g. in the `gabls1` example.

**Reply by makrandak** [reply]
Hello, 

I am looking into planarAvg; thanks for the help. 

Is there an example where the entire spatial average (in all directions is computed)? I want to subtract the average of a field at every timestep from itself after every timestep. 

Thank you for your time and help. 

With sincere regards,
Makrand
